src/peer-store/README.md

```diff
@@ -1,3 +1,156 @@
+
+It is also important pointing out that users would be able to use `peerStore.protoBook.*`, so eventually we should move those into `peerStore._ protoBook.*` if we do not intend them to use it.
+
+---
+
+- `peerStore.get(peerId, options)`
+
+Get the information of a provided peer. The amount of information that we want can be customized with the following options, which are true by default:
+
+```js
+{
+  address: true,
+  proto: true,
+  key: true,
+  metadata: true
+}
+```
+
+- `peerStore.delete(peerId, [data])`
+
+Deletes the provided peer from every book. If data is provided, just remove the data from the books. The data should be provided as follows:
+
+```js
+{
+  address: [],
+  proto: [],
+  key: [],
+  metadata: []
+}
+```
+
+- `peerStore.peers(options)`
+
+Get an array of all the peers, as well as their information. The information intended can be customized with the following options, which are true by default:
+
+```js
+{
+  address: true,
+  proto: true,
+  key: true,
+  metadata: true
+}
+```
+
+## Address Book
+
+The `addressBook` keeps the known multiaddrs of a peer. The multiaddrs of each peer are not a constant and the Address book must have this into consideration.
+
+`Map<string, multiaddrInfo>`
+
+A `peerId.toString()` identifier mapping to a `multiaddrInfo` object, which should have the following structure:
+
+```js
+{
+  multiaddr: ,
+  validity: ,
+  confidence: 
+}
+```
+
+**Note:** except for multiaddr namings, the other properties are placeholders for now and might not be as described in the future milestones.
+
+- `addressBook.set()`
+- `addressBook.get()`
+- `getMultiaddrsForPeer()`
+- `addressBook.has()`
+- `addressBook.delete()`
+- `addressBook.peers()`
+
+It is important pointing out that the API methods which return arrays of data (`set`, `get`, `getMultiaddrsForPeer`) shuld return the `multiaddr` property of the `multiaddrInfo` and not the entire `multiaddrInfo` as the remaining data should be used internally. Should we consider having two datastructure instead?
+
+Further API methods will probably be added in the context of multiaddr `ttl` and multiaddr confidence.
+
+**Not Yet Implemented**: Multiaddr Confidence
+
+## Key Book
+
+The `keyBook` tracks the keys of the peers.
+
+**Not Yet Implemented**
+
+## Protocol Book
+
+The `protoBook` holds the identifiers of the protocols supported by each peer. The protocols supported by each peer are dynamic and will change over time.
+
+`Map<string, Set<string>>`
+
+A `peerId.toString()` identifier mapping to a `Set` of protocol identifier strings.
+
+- `protoBook.set()`
+- `protoBook.get()`
+- `protoBook.has()`
+- `protoBook.delete()`
+- `protoBook.supports()`
```

Comment:
Right now it looks like `has` is answering "do I have protocols for this peer?", but that doesn't seem actionable to me. I don't know why I'd ever ask that question. `supports` makes sense, because I will want to check protocol support before doing certain actions.

These Books aren't Map interface compliant so I don't see a need to add methods unnecessarily if they don't make sense.

Reply:
`has(peerId)` would check if we have protocols for that peer, while `supports(peerId, protocol)` would say if it supports the peer. So, it is a supplement from has. Anyway, we can achieve the `has` with `get`, so we can drop it if you feel that there is no use case where this is useful.

Reply:
Yeah I think we should drop it. I expect users to either want to look at all the protocols for a peer `.get`, or (and the likely thing) check for specific support for a set of protocols `.supports`.